source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -764,6 +776,13 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
   typed_tls_data.context_ = &json_body;
   typed_tls_data.environ_ = &environ_;
   typed_tls_data.cluster_metadata_ = cluster_metadata;
+  typed_tls_data.dynamic_metadata_ = dynamic_metadata;
+
+  // If there is a value we're in a upstream filter
+  if (callbacks.upstreamCallbacks().has_value()) {
```

Comment:
`typed_tls_data.endpoint_metadata_` **must** be set to nil if not set to upstrema host metadata; as otherwise endpoint_metadata_ may "leak" 

maybe follow the example of how `dynamic_metadata` is set? create a variable that's nill; try to fill it, and in the end set it anyway in the typed_tls_data? this guarantees that it is always set to correct value

Reply:
Sounds good, will update